src/utils/hooks/useUpdateReviewResponseNEW.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  ReviewResponse,
+  ReviewResponseDecision,
+  ReviewResponseRecommendedApplicantVisibility,
+  useUpdateReviewResponseMutation,
+} from '../generated/graphql'
+
+// below lines are used to get return type of the function that is returned by useUpdateReviewResponseMutation
+type UseUpdateReviewMutationReturnType = ReturnType<typeof useUpdateReviewResponseMutation>
+type promiseReturnType = ReturnType<UseUpdateReviewMutationReturnType[0]>
+
+// hook used to update review response, comment and decision, use as per type definition belo
+// also computes and updates recommendedApplicantVisibility
+type useUpdateReviewResponse = (
+  reviewResponseId: number
+) => (reviewResponse: ReviewResponse) => promiseReturnType
+
+const useUpdateReviewResponse: useUpdateReviewResponse = (reviewResponseId) => {
+  const [updateReviewResponse] = useUpdateReviewResponseMutation()
+
+  return async (reviewResponse) =>
+    updateReviewResponse({
+      variables: {
+        id: reviewResponseId,
+        comment: reviewResponse.comment,
+        decision: reviewResponse.decision,
+        recommendedApplicantVisibility: computeVisibility(reviewResponse),
+      },
+    })
+}
+
+// is_visible_to_applicant is set by an action on back, it uses recomended_application_visibility of the
+// last level review to determine what will be visible to application in LOQ
+// we can add a checkbox in decision modal to determine this (thus logic is placed here)
+const computeVisibility = (reviewResponse: ReviewResponse | undefined) => {
+  // would rather computer this from actual review level, but dont' want to pass too far as props or rely on context
+  if (!reviewResponse)
+    return ReviewResponseRecommendedApplicantVisibility.OriginalResponseNotVisibleToApplicant
+
+  const isLevelOne = reviewResponse.originalReviewResponseId === reviewResponse.id
+
+  if (isLevelOne) {
+    if (reviewResponse.decision === ReviewResponseDecision.Decline)
+      return ReviewResponseRecommendedApplicantVisibility.OriginalResponseVisibleToApplicant
+
+    return ReviewResponseRecommendedApplicantVisibility.OriginalResponseNotVisibleToApplicant
+  }
+  // consolidator
+  if (
+    reviewResponse.decision === ReviewResponseDecision.Agree &&
+    reviewResponse.originalReviewResponse?.decision === ReviewResponseDecision.Decline
+  )
+    return ReviewResponseRecommendedApplicantVisibility.OriginalResponseVisibleToApplicant
```

Comment:
Just re-iterating the comment I made elsewhere. If the original reviewer has made a comment, but Approved, shouldn't this be visible to applicant as well? So maybe this should check for a non-empty comment on originalReviewResponse rather than "Declined" decision?

Reply:
I've made a seperate comment about it, hope it makes sense? https://github.com/openmsupply/application-manager-web-app/pull/425#issuecomment-799824628